---
title: "Your Title"
subtitle: "BMIN503/EPID600 Final Project"
author: "FirstName LastName"
format: html
editor: visual
number-sections: true
embed-resources: true
---

------------------------------------------------------------------------

Use this template to complete your project throughout the course. Your Final Project presentation will be based on the contents of this document. Replace the title/name above and text below with your own, but keep the headers. Feel free to change the theme and other display settings, although this is not required. I added a new sentence

## Overview {#sec-overview}

This project aims to simplify finding good gene therapy candidates based on certain parameters. While focusing on AAV based gene therapy, the code will also be usable for other existing and future vectors such as lentivirus and liquid nanoparticle (LNP).

## Introduction {#sec-introduction}

Genetic diseases are health problems caused by defects in an individuals DNA. Because of this, most medicines and treatments only offer short term relief as the core issue with the DNA remains. In recent history, the rise of gene therapy has given hope to those with genetic diseases in the form of a long lasting cure by providing functional DNA via different vectors. The drawbacks of gene therapy include both the limitations of vectors as well as the long development time of treatments. Because of this, selecting what disease to develop drugs for is a critical step to save time and resources. This code works to provide an important resource to navigate through the thousand of genetic diseases in order to increase efficiency in diseases we can cure.

Being in the field of medicine, gene therapy is an interdisciplinary field. Ranging from doctors identifying diseases, geneticists sequencing genes, molecular biologists creating vectors, veterinarians overseeing animal trials, data scientists creating databases, and the many government agencies that oversee this, there are multiple fields that can contribute to working on this issue. Important data for this issue includes disease tropism, disease prevalence, and gene size, which can be found in various databases kept by different agencies.

## Methods {#sec-methods}

Different databases were explored and used for data extraction. More databases will be explored.

```{r}
#OMIM (Online Mendelian Inheritance in Man org) database extracted files

#This file provides links between the genes in OMIM and other gene identifiers
omim1 <- read.table("https://omim.org/static/omim/data/mim2gene.txt", header = FALSE, sep = "\t")
colnames(omim1) <- c("MIM Number", "MIM Entry Type", "Entrez Gene ID (NCBI)", "Approved Gene Symbol (HGNC)", "Ensembl Gene ID (Ensembl)")
head(omim1)

#Table of MIM numbers and titles
omim2 <- read.table("C:\\Users\\Hasaj\\OneDrive\\Documents\\BMIN\\BMIN503\\BMIN503_Final_Project\\omim2.txt", header = FALSE, fill =TRUE, sep = "\t")
colnames(omim2) <- c("Prefix", "MIM Number", "Preferred Title", "Symbols", "Notes")
head(omim2)

#OMIM Synopsis, genomic coordiantes, and inheritance
omim3 <- read.table("C:\\Users\\Hasaj\\OneDrive\\Documents\\BMIN\\BMIN503\\BMIN503_Final_Project\\omim3.txt", header = FALSE, fill= TRUE, sep = "\t")
colnames(omim3) <- c("Chromosome", "Genomic Position Start", "Genomic Position End", "Cyto Location", "Computed Cyto Location", "MIM Number", "Gene/Locus and Related Symbols", "Gene Name", "Approved Gene Symbol", "Entrez Gene ID", "Ensembl Gene ID",  "Comments", "Phenotypes", "Mouse Gene Symbol/ID")
head(omim3)


#Sorted table of OMIM synopsis by disorder
omim4 <-read.table("C:\\Users\\Hasaj\\OneDrive\\Documents\\BMIN\\BMIN503\\BMIN503_Final_Project\\omim4.txt", header = FALSE, fill= TRUE, sep = "\t")
colnames(omim4) <- c("Phenotype", "Gene/Locus symbols", "MIM Number", "Cyto Location")
head(omim4)


```

For the purpose of this study, omim3 is the most important table as it has information on gene length. This table will be transformed to find the length of the genes for later use.

```{r}
library(dplyr)

#Create a table with gene length and populate it with data
class (omim3$`Genomic Position Start`)

#First convert character to numeric, then add new column with data

omim3.gl <- omim3 %>%
  mutate(
    `Genomic Position End` = as.numeric(`Genomic Position End`),
    `Genomic Position Start` = as.numeric(`Genomic Position Start`),
    Gene.Length = `Genomic Position End` - `Genomic Position Start`
  )


```

## Results {#sec-results}

With cleaned data, searches for results that fit desired criteria can be made.

To still be included is the joining (left join) of other OMIM tables to add additional information about potential diseases as well as tables from other databases. Additional support for other vectors to be added.

```{r}

#AAV ideal works with gene sizes below 4.5kb
#Sort data by genes smaller than 4.5kb. The smaller the gene the more likely successful transduction happens, so sorting by size is helpful
omim3.gl.aav <- omim3.gl %>%
    filter(Gene.Length < 4500) %>%
    arrange(Gene.Length)
head(omim3.gl.aav)
tail(omim3.gl.aav)

#Liquid nanoparticle vectors typically can handle 8-10 kbp DNA inserts
omim3.gl.lnp <- omim3.gl %>%
    filter(Gene.Length < 10000) %>%
    arrange(Gene.Length)

#Lenti Viral vectors typically can handle 6-8 kbp DNA inserts
omim3.gl.lv <- omim3.gl %>%
    filter(Gene.Length < 8000) %>%
    arrange(Gene.Length)


omim3.gl %>% filter(`Approved Gene Symbol` == "ANON1")

```

## Conclusion

This the conclusion. The @sec-results can be invoked here.
